tikv/tikv
# engine: enable static_libcpp feature of rust-rocksdb

### What problem does this PR solve?
After https://github.com/tikv/rust-rocksdb/pull/463 and https://github.com/tikv/rust-rocksdb/pull/464 static linking with libc++ become optional. To make tikv static link with libc++, we need to enable the `static_libcpp` feature of rust-rocksdb.

### What is changed and how it works?
enable static_libcpp feature of rust-rocksdb

### Check List 

Tests 
- Manual test (add detailed scripts or steps below)
build tikv locally and run `ldd tikv-server` to check if libc++ is dynamic linked.

update rust-rocksdb without setting `static_libcpp`:
```
root@docker:/data/tikv > ldd target/release/tikv-server
        linux-vdso.so.1 (0x00007ffe65dfb000)
        libstdc++.so.6 => /lib/x86_64-linux-gnu/libstdc++.so.6 (0x00007f96510dd000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f96510ba000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f96510b4000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f9651099000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f9650ea7000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f9653d1c000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f9650d58000)
```

update rust-rocksdb and set `static_libcpp`:
```
root@docker:/data/tikv > ldd target/release/tikv-server
        linux-vdso.so.1 (0x00007ffe6036d000)
        libpthread.so.0 => /lib/x86_64-linux-gnu/libpthread.so.0 (0x00007f6f7faef000)
        libdl.so.2 => /lib/x86_64-linux-gnu/libdl.so.2 (0x00007f6f7fae9000)
        libgcc_s.so.1 => /lib/x86_64-linux-gnu/libgcc_s.so.1 (0x00007f6f7face000)
        libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f6f7f8dc000)
        /lib64/ld-linux-x86-64.so.2 (0x00007f6f825ea000)
        libm.so.6 => /lib/x86_64-linux-gnu/libm.so.6 (0x00007f6f7f78d000)
```
